File: Dev/C++/Mascaret/src/VEHA/Behavior/BasicActions/CallBehaviorAction.cpp
#include "VEHA/Behavior/BasicActions/CallBehaviorAction.h"
#include "VEHA/Behavior/BasicActions/CallBehaviorBehaviorExecution.h"
#include "VEHA/Kernel/Operation.h"
#include "VEHA/Kernel/Parameter.h"
#include <boost/algorithm/string.hpp>

namespace VEHA
{

CallBehaviorAction::CallBehaviorAction()
{
	_kind = "CallBehavior";
}

CallBehaviorAction::~CallBehaviorAction()
{
}

shared_ptr<BehaviorExecution> CallBehaviorAction::createBehaviorExecution(shared_ptr<InstanceSpecification> host, const Parameters& p, bool sync)
{
	shared_ptr<CallBehaviorBehaviorExecution> behavior(new CallBehaviorBehaviorExecution(shared_dynamic_cast<CallBehaviorAction>(shared_from_this()),host,p));
	return behavior;
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Behavior/BasicActions/CallBehaviorAction.cs
using System;
using System.Collections.Generic;

namespace Mascaret
{
    public class CallBehaviorAction : Action, IEquatable<CallBehaviorAction>
    {

        private InstanceSpecification target;
        public InstanceSpecification Target
        {
            get { return target; }
            set { target = value; }
        }

        private string stringArguments;
        public string StringArguments
        {
            get { return stringArguments; }
            set { stringArguments = value; }
        }

        private Behavior behavior;
        public Behavior Behavior
        {
            get { return behavior; }
            set { behavior = value; }
        }

        public CallBehaviorAction()
        {
            Kind = "CallBehavior";
        }

        public bool Equals(CallBehaviorAction other)
        {
            if (other == this)
                return true;
            return false;
        }

        public override BehaviorExecution createBehaviorExecution(InstanceSpecification host, Dictionary<string, ValueSpecification> p, bool sync)
        {
            CallBehaviorBehaviorExecution behavior = new CallBehaviorBehaviorExecution(this, host, p);
            return behavior;
        }
    }
}